src/execution/sema/sema_builtin.cpp

```diff
@@ -0,0 +1,1878 @@
+#include "execution/sema/sema.h"
+
+#include "execution/ast/ast_node_factory.h"
+#include "execution/ast/context.h"
+#include "execution/ast/type.h"
+
+namespace terrier::execution::sema {
+
+namespace {
+
+bool IsPointerToSpecificBuiltin(ast::Type *type, ast::BuiltinType::Kind kind) {
+  if (auto *pointee_type = type->GetPointeeType()) {
+    return pointee_type->IsSpecificBuiltin(kind);
+  }
+  return false;
+}
+
+bool IsPointerToSQLValue(ast::Type *type) {
+  if (auto *pointee_type = type->GetPointeeType()) {
+    return pointee_type->IsSqlValueType();
+  }
+  return false;
+}
+
+bool IsPointerToAggregatorValue(ast::Type *type) {
+  if (auto *pointee_type = type->GetPointeeType()) {
+    return pointee_type->IsSqlAggregatorType();
+  }
+  return false;
+}
+
+template <typename... ArgTypes>
+bool AreAllFunctions(const ArgTypes... type) {
+  return (true && ... && type->IsFunctionType());
+}
+
+}  // namespace
+
+void Sema::CheckBuiltinMapCall(UNUSED ast::CallExpr *call) {}
+
+void Sema::CheckBuiltinSqlConversionCall(ast::CallExpr *call, ast::Builtin builtin) {
+  if (builtin == ast::Builtin::DateToSql) {
```

Comment:
The type in `value.h` and its functions are tested.
The builtin calls `@dateToSql` itself was tested `.tpl` files that are now removed (they depend on the CSV reader).
It is not in the switch statement because it takes a different number of arguments (3 instead of 1 for year, month, day).